sdks/java/extensions/python/src/main/resources/org/apache/beam/sdk/extensions/python/bootstrap_beam_venv.py

```diff
@@ -92,6 +92,17 @@ def maybe_strict_version(s):
     if not os.path.exists(venv_python):
         try:
             subprocess.run([executable, '-m', 'venv', venv_dir], check=True)
+
+            # Upgrading pip and setuptools for the virtual environment.
+            subprocess.run([
+                venv_python, '-m', 'pip', 'install', '--upgrade', 'pip'
```

Comment:
Would this two additional installations increase the launch time? Was there any noticeable difference?

Reply:
I didn't notice a considerable increase in the launch time.